server/src/pages/shot/editor.js

```diff
@@ -213,319 +144,64 @@ exports.Editor = class Editor extends React.Component {
 
   onClickCrop() {
     this.previousTool = this.state.tool;
-    this.setState({tool: "crop"});
-    this.cropToolBar = <div className="editor-header default-color-scheme"><div className="annotation-tools">
-      <Localized id="annotationCropConfirmButton">
-        <button className={`button transparent confirm-crop`} id="confirm-crop" onClick={this.onClickConfirmCrop.bind(this)} title="Confirm selection">Crop</button>
-      </Localized>
-      <Localized id="annotationCropCancelButton">
-        <button className={`button transparent cancel-crop`} id="cancel-crop" onClick={this.onClickCancelCrop.bind(this)} title="Cancel selection">Cancel</button>
-      </Localized>
-    </div></div>;
+    this.setState({tool: "cropTool"});
     sendEvent("crop-select", "annotation-toolbar");
   }
 
-  onClickConfirmCrop() {
-    const x1 = Math.max(selectedPos.left, 0);
-    const x2 = Math.min(selectedPos.right, this.canvasWidth);
-    const y1 = Math.max(selectedPos.top, 0);
-    const y2 = Math.min(selectedPos.bottom, this.canvasHeight);
-    const cropWidth = Math.floor(x2 - x1);
-    const cropHeight = Math.floor(y2 - y1);
-    if (!selectedPos.width || !selectedPos.height || (this.canvasHeight === cropHeight) && (this.canvasWidth === cropWidth)) {
-      this.removeCropBox();
-      this.cropToolBar = null;
+  onDrawingUpdate(affectedArea, incomingCanvas) {
+    if (!affectedArea || !incomingCanvas) {
+      return;
+    }
+
+    this.imageContext.drawImage(incomingCanvas,
+      affectedArea.left * this.devicePixelRatio,
+      affectedArea.top * this.devicePixelRatio,
+      affectedArea.width * this.devicePixelRatio,
+      affectedArea.height * this.devicePixelRatio,
+      affectedArea.left, affectedArea.top, affectedArea.width, affectedArea.height);
+  }
```

Comment:
This saves an edit over a certain area of the image.  For tracking changes for undo/redo, we can first copy the affected area of the image into another image before applying the edit.